wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -144,16 +144,25 @@ public void validate(Environment environment)
    *
    * @param workspace the workspace to inject runtime into
    */
-  public void injectRuntime(WorkspaceImpl workspace) {
+  public void injectRuntime(WorkspaceImpl workspace) throws ServerException {
```

Comment:
I'd rather use InfrastructureException. We have discussed that ServerException if rather REST exception.

Reply:
I agree that `ServerException` is REST exception, but `InfrastructureException` is an exception that is supposed to be thrown by Infrastructure. As you can see, now `WorkspaceRuntimes` throw only REST exceptions (Conflict, NotFound, Server).
I'd say that WorkspaceRuntimes should not throw REST exception but and Infrastructure exception isn't the right one here too. I guess there is an issue for reworking Che Master exception approach and this issue could be solved there.
@garagatyi Does it makes sense?

Reply:
Yep, you are completely right. I missed that.